R/utils-list.R

```diff
@@ -34,28 +34,40 @@ strip_pts <- function(list, param){
   for (v in param){
     if (v %in% names(list) && !inherits(list[[v]], c('function','formula'))) {
       v.vals <- list[[v]]
-      out <- append(out, v.vals)
-      out.class <- ifelse(!all(is.na(v.vals)), class(v.vals), out.class)
+      out <- append_keepTZ(out, v.vals, tz=attr(v.vals, 'tzone'))
+      all.na <- all(is.na(v.vals))
+      out.class <- ifelse(!all.na, class(v.vals), out.class)
+      if(all.na){ 
+        class(out) <- out.class 
+      }
     } else {
       if (any(sapply(list, is.list))){
         u.list <- unname_c(list[sapply(list, is.list)])
         if(v %in% names(u.list)) {
           v.vals <- u.list[[v]]
-          out <- append(out, v.vals)
-          out.class <- ifelse(!all(is.na(v.vals)), class(v.vals), out.class)
+          out <- append_keepTZ(out, v.vals, tz=attr(v.vals, 'tzone'))
+          all.na <- all(is.na(v.vals))
+          out.class <- ifelse(!all.na, class(v.vals), out.class)
+          if(all.na){ 
+            class(out) <- out.class 
+          }
         } else if (any(sapply(u.list, function(x) any(names(x) %in% v)))) {
           v.vals <- u.list[[which(sapply(u.list, function(x) any(names(x) %in% v)))]][[v]]
-          out <- append(out, v.vals)
-          out.class <- ifelse(!all(is.na(v.vals)), class(v.vals), out.class)
+          out <- append_keepTZ(out, v.vals, tz=attr(v.vals, 'tzone'))
+          all.na <- all(is.na(v.vals))
+          out.class <- ifelse(!all.na, class(v.vals), out.class)
+          if(all.na){ 
+            class(out) <- out.class 
+          }
         } else {
-          out <- append(out, NA)
+          out <- append_keepTZ(out, NA, tz=attr(out, "tzone"))
```

Comment:
is it not necessary to set the class here too? Just wondering what cases we lose it for, and if this is a safe one